homeassistant/components/sonos/speaker.py

```diff
@@ -283,25 +283,23 @@ def share_link(self) -> ShareLinkPlugin:
         return self._share_link_plugin
 
     @property
-    def subscription_address(self) -> str | None:
+    def subscription_address(self) -> str:
         """Return the current subscription callback address if any."""
-        if self._subscriptions:
-            addr, port = self._subscriptions[0].event_listener.address
-            return ":".join([addr, str(port)])
-        return None
+        addr, port = self._subscriptions[0].event_listener.address
+        return ":".join([addr, str(port)])
 
     #
     # Subscription handling and event dispatchers
     #
     def log_subscription_result(
-        self, result: Any, event: str, level: str = logging.DEBUG
+        self, result: Any, event: str, level: int = logging.DEBUG
     ) -> None:
         """Log a message if a subscription action (create/renew/stop) results in an exception."""
         if not isinstance(result, Exception):
             return
 
         if isinstance(result, asyncio.exceptions.TimeoutError):
-            message = "Request timed out"
+            message: Exception | str = "Request timed out"
             exc_info = None
         else:
             message = result
```

Comment:
```suggestion
            message = "Request timed out"
            exc_info = None
        else:
            message = str(result)
```
The exception is string formatted for the log output anyway. Might as well do it here.